api/service.py

```
@@ -173,6 +198,37 @@ async def lifespan(app: FastAPI):
+            redis_client.redis,
+            "api",
+            rate_configs["api"][0],  # rate_per_second
+            rate_configs["api"][1]   # burst_capacity
+        )
+
+        # Add rate limiting middleware after initialization
+        if telemetry_limiter and api_limiter:
+            app.add_middleware(RateLimitMiddleware, telemetry_limiter=telemetry_limiter, api_limiter=api_limiter)
+
+        print("✅ Rate limiting initialized successfully")
+    except Exception as e:
```

Ruff (0.14.10)

Do not catch blind exception: `Exception`

(BLE001)